src/main/java/org/broadinstitute/hellbender/tools/walkers/bqsr/ApplyBQSR.java

```diff
@@ -47,18 +47,23 @@
     
     private SAMFileGATKReadWriter outputWriter;
 
-    private ReadTransformer transform;
+    /**
+     * Returns the BQSR transformer
+     */
+    @Override
+    public ReadTransformer makeReadTransformer(){
+        return new BQSRReadTransformer(getHeaderForReads(), BQSR_RECAL_FILE, bqsrArgs);
+    }
 
     @Override
     public void onTraversalStart() {
         outputWriter = createSAMWriter(OUTPUT, true);
-        transform = new BQSRReadTransformer(getHeaderForReads(), BQSR_RECAL_FILE, bqsrArgs);
         Utils.warnOnNonIlluminaReadGroups(getHeaderForReads(), logger);
     }
 
     @Override
     public void apply( GATKRead read, ReferenceContext referenceContext, FeatureContext featureContext ) {
-        outputWriter.addRead(transform.apply(read));
+        outputWriter.addRead(read);
```

Comment:
We could just leave bqsr to apply the transformer explicitly call transform manually during apply.